server/service/src/repack/generate.rs

```diff
@@ -0,0 +1,204 @@
+        on_hold: false,
+        created_datetime: Utc::now().naive_utc(),
+        verified_datetime: Some(Utc::now().naive_utc()),
+        ..Default::default()
+    };
+
+    let mut invoice_lines = Vec::new();
+
+    let item = ItemRowRepository::new(connection)
+        .find_one_by_id(&stock_line.item_id)?
+        .ok_or(RepositoryError::NotFound)?;
+
+    let stock_in = InvoiceLineRow {
+        id: uuid(),
+        invoice_id: invoice.id.clone(),
+        item_id: stock_line.item_id.clone(),
+        item_name: item.name.clone(),
+        item_code: item.code.clone(),
+        stock_line_id: Some(new_stock_line.id.clone()),
+        location_id: new_stock_line.location_id.clone(),
+        batch: stock_line.batch.clone(),
+        expiry_date: stock_line.expiry_date,
+        pack_size: new_stock_line.pack_size,
+        r#type: InvoiceLineRowType::StockIn,
+        number_of_packs: new_stock_line.total_number_of_packs,
+        cost_price_per_pack: new_stock_line.cost_price_per_pack,
+        sell_price_per_pack: new_stock_line.sell_price_per_pack,
+        ..Default::default()
+    };
+
+    let stock_out = inline_edit(&stock_in, |mut l| {
+        l.id = uuid();
+        l.stock_line_id = Some(stock_line.id.clone());
+        l.location_id = stock_line.location_id.clone();
+        l.pack_size = stock_line.pack_size;
+        l.r#type = InvoiceLineRowType::StockOut;
+        l.number_of_packs = new_stock_line.pack_size as f64 * new_stock_line.total_number_of_packs;
+        l.cost_price_per_pack = stock_line.cost_price_per_pack;
+        l.sell_price_per_pack = stock_line.sell_price_per_pack;
+
+        l
+    });
+
+    invoice_lines.push(stock_in);
+    invoice_lines.push(stock_out);
+
+    Ok((invoice, invoice_lines))
+}
+
+fn generate_new_stock_lines(stock_line: &StockLineRow, input: &InsertRepack) -> StockLineJob {
+    let stock_line_total = calculate_stock_line_total(stock_line);
+
+    let stock_line_to_update = if stock_line_total == input.number_of_packs {
+        let mut new_line = stock_line.clone();
+
+        new_line.available_number_of_packs = 0.0;
+        new_line.total_number_of_packs = 0.0;
+
+        new_line
+    } else {
+        let mut new_line = stock_line.clone();
+
+        new_line.available_number_of_packs =
+            new_line.available_number_of_packs - input.number_of_packs;
+        new_line.total_number_of_packs = new_line.total_number_of_packs - input.number_of_packs;
+
+        new_line
+    };
+
+    let new_stock_line = {
+        let mut new_line = stock_line.clone();
+
+        new_line.id = uuid();
+        new_line.pack_size = input.new_pack_size;
+        new_line.available_number_of_packs =
+            input.number_of_packs * stock_line.pack_size as f64 / input.new_pack_size as f64;
+        new_line.total_number_of_packs =
+            input.number_of_packs * stock_line.pack_size as f64 / input.new_pack_size as f64;
+
+        if stock_line.pack_size > input.new_pack_size {
```

Comment:
isn't it just:

cost_price_per_pack = cost_price_per_pack * (input.new_pack_size/stock_line.pack_size)

![Screenshot 2023-05-10 at 1 00 11 PM](https://github.com/openmsupply/open-msupply/assets/26194949/32508bdb-cb86-4f28-8a0e-81e0e9e71907)